mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -786,9 +824,14 @@ static void cleanUpDeadVals(RDVFinalCleanupList &list) {
 
   // 3. Operations
   LDBG() << "Cleaning up " << list.operations.size() << " operations";
-  for (auto &op : list.operations) {
+  for (Operation *op : list.operations) {
     LDBG() << "Erasing operation: "
            << OpWithFlags(op, OpPrintingFlags().skipRegions());
+    if (op->hasTrait<OpTrait::IsTerminator>()) {
+      // When erasing a terminator, insert an unreachable op in its place.
+      OpBuilder b(op);
+      ub::UnreachableOp::create(b, op->getLoc());
+    }
```

Comment:
```suggestion
    // When erasing a terminator, insert an unreachable op in its place.
    if (op->hasTrait<OpTrait::IsTerminator>())
      ub::UnreachableOp::create(OpBuilder{op}, op->getLoc());
```

Nit: simplifying a bit.